qlib/data/data.py

```diff
@@ -517,18 +552,8 @@ def load_calendar(self, freq, future):
         list
             list of timestamps
         """
-        if future:
-            fname = self._uri_cal.format(freq + "_future")
-            # if future calendar not exists, return current calendar
-            if not os.path.exists(fname):
-                get_module_logger("data").warning(f"{freq}_future.txt not exists, return current calendar!")
-                fname = self._uri_cal.format(freq)
-        else:
-            fname = self._uri_cal.format(freq)
-        if not os.path.exists(fname):
-            raise ValueError("calendar not exists for freq " + freq)
-        with open(fname) as f:
-            return [pd.Timestamp(x.strip()) for x in f]
+        self.backend.setdefault("kwargs", {}).update(freq=freq, future=future)
+        return [pd.Timestamp(x) for x in self.backend_obj.data]
```

Comment:
If we have to set the config for each backend_obj,  why don't use the config as the parameters of the function?

Reply:
self.backend_obj(freq=freq, future=future).data